fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -341,14 +343,51 @@ private void pruneTables(OptExpression tree, TaskContext rootTaskContext, Column
             ruleRewriteIterative(tree, rootTaskContext, new MergeTwoProjectRule());
             rootTaskContext.setRequiredColumns(requiredColumns.clone());
             ruleRewriteOnlyOnce(tree, rootTaskContext, RuleSetType.PRUNE_COLUMNS);
-            context.setEnableLeftRightJoinEquivalenceDerive(true);
             ruleRewriteIterative(tree, rootTaskContext, RuleSetType.PUSH_DOWN_PREDICATE);
         }
     }
 
+    private void ruleBasedMaterializedViewRewriteMultiStages(OptExpression tree,
+                                                             TaskContext rootTaskContext,
+                                                             ColumnRefSet requiredColumns) {
+        if (!mvRewriteStrategy.mvStrategy.isMultiStages()) {
+            return;
+        }
+
+        OptimizerTraceUtil.logOptExpression("before RuleBasedMaterializedViewRewrite:\n%s", tree);
```

Comment:
Good trace. But I think we should also consider ruleBasedMaterializedViewRewrite, should we distinguish these two stages? add more clear debug infos.
I think we would better add trace log in ruleBasedMaterializedViewRewrite.